tentacle/src/transports/tls.rs

```diff
@@ -0,0 +1,257 @@
+use super::Result;
+use futures::{future::ok, SinkExt, Stream, TryFutureExt};
+use log::warn;
+use std::{
+    future::Future,
+    net::SocketAddr,
+    pin::Pin,
+    task::{Context, Poll},
+    time::Duration,
+};
+
+use crate::runtime::TcpListener;
+use crate::transports::TransportFuture;
+use crate::{
+    error::TransportErrorKind,
+    multiaddr::Multiaddr,
+    runtime::TcpStream,
+    transports::{tcp_dial, tcp_listen, Transport},
+    utils::{dns::DnsResolver, multiaddr_to_socketaddr, socketaddr_to_multiaddr},
+};
+use futures::channel::mpsc::{channel, Receiver, Sender};
+use std::sync::Arc;
+use tokio::io;
+use tokio_rustls::rustls::{ClientConfig, ServerConfig};
+use tokio_rustls::webpki::DNSNameRef;
+use tokio_rustls::{client, server, TlsAcceptor, TlsConnector};
+
+pub type TlsServerStream = Box<server::TlsStream<TcpStream>>;
```

Comment:
Now that the box is already boxed, why not merge into `Box<dyn AsyncRW + Send>` if the behavior is the same